### PR TITLE
feat: add a new plugin finance news

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new plugin finance news

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: plugin/bundles/finance_news/plugins/finance_news_search/plugin.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class FinanceNewsSearch(PluginHandler):
+    async def execute(self, credentials: BundleCredentials, plugin_input: PluginInput) -> PluginOutput:
+        date: str = plugin_input.input_params.get("date", None)
+        keyword: str = plugin_input.input_params.get("keyword", None)
+        limit: int = plugin_input.input_params.get("limit", 25)
+        source: str = plugin_input.input_params.get("source", None)
+        tags: str = plugin_input.input_params.get("tags", None)
+        tickers: str = plugin_input.input_params.get("tickers", None)
+
+        finance_news_api_key: str = credentials.credentials.get("FINANCE_NEWS_API_KEY")
```
Comment (DynamesC) [inline plugin/bundles/finance_news/plugins/finance_news_search/plugin.py:19]: check input data formats before using them

File: plugin/bundles/finance_news/resources/i18n/en.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+bundle_name: "Finance News"
+bundle_description: "Real-time Finance News API using the JSON payload to extract dynamic financial data."
+credential_finance_news_api_key_description: "API Key for authentication. Get it from https://apilayer.com/marketplace/financelayer-api"
+finance_news_search_input_date_name: "Date"
+finance_news_search_input_date_description: "Use this parameter to specify a date or date range. Example: &date=2020-01-01 for news on Jan 1st and &date=2020-12-24,2020-12-31 for news between Dec 24th and 31st. Available options: today, thisweek, thismonth, last7days, last30days, last3months, yeartodate, YYYY-MM-DD"
```
Comment (DynamesC) [inline plugin/bundles/finance_news/resources/i18n/en.yml:5]: 1. remove '&date='
2. Make descriptions starting from 'Available options' more readable.

File: plugin/bundles/finance_news/resources/i18n/en.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+finance_news_search_input_keyword_name: "Keyword"
+finance_news_search_input_keyword_description: "Keyword to search. Example: &keyword=bitcoin"
+finance_news_search_input_limit_name: "Limit"
+finance_news_search_input_limit_description: "Use this parameter to specify a pagination limit (number of results per page) for your API request. The default limit value is 25, the maximum allowed limit value is 100."
+finance_news_search_input_source_name: "Source"
+finance_news_search_input_source_description: "Use this parameter to include or exclude one or multiple comma-separated news sources. Example: To include forbes.com, but exclude bloomberg.com: &sources=forbes.com, -bloomberg.com"
```
Comment (DynamesC) [inline plugin/bundles/finance_news/resources/i18n/en.yml:11]: remove '&sources'

File: plugin/bundles/finance_news/resources/i18n/en.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+finance_news_search_input_tags_name: "Tags"
+finance_news_search_input_tags_description: "Use this parameter to search for tags by which news are tagged. You can search by multiple tags, separating tags with a comma, you can also exclude tags that you do not want to appear in your search results. Example: &tags=Bitcoin, -money. Results will exclude tags money from results"
```
Comment (DynamesC) [inline plugin/bundles/finance_news/resources/i18n/en.yml:13]: remove '&tags'

**Review by DynamesC — CHANGES_REQUESTED** [review]

**Review by DynamesC — APPROVED** [review]
LGTM

**Review by jameszyao — APPROVED** [review]
LGTM

**Pull request merged**